.github/workflows/setup-cf-env.yml

```diff
@@ -0,0 +1,95 @@
+        name: Claim Toolsmiths Environment
+        env:
+          api_token: ${{ secrets.TOOLSMITHS_API_TOKEN }}
+          hostname:  ${{ secrets.TOOLSMITHS_HOSTNAME }}
+          notes:     CF CLI Github Actions Integration Tests
+          pool_name: cf-deployment
+        run: |
+          while true; do
+            curl -s --show-error -D >(tee headers.txt >&2) -H 'Accept: application/json' \
+            -X POST "https://${hostname}/pooled_gcp_engineering_environments/claim" \
+            --data-urlencode "api_token=${api_token}" \
+            --data-urlencode "pool_name=${pool_name}" \
+            --data-urlencode "notes=${notes}" > metadata.json \
+            || echo "Unable to reach server, trying again in 30 seconds..."
+
+            ERR_500="Sorry, the Toolsmiths Environments app is currently encountering issues. Trying again in 30 seconds..."
+            ERR_429="Sorry, Toolsmiths are out of environments in your requested pool. New environments are on their way but you can stop by the Toolsmiths slack channel for more help."
+            ERR_409="Sorry, was not able to claim an environment. Trying again in 30 seconds..."
+
+            grep -q -E "HTTP/[[:digit:]\.]{1,3} 401" headers.txt && exit 1
+            grep -q -E "HTTP/[[:digit:]\.]{1,3} 404" headers.txt && exit 2
+            grep -q -E "HTTP/[[:digit:]\.]{1,3} 500" headers.txt && echo "$ERR_500"
+            grep -q -E "HTTP/[[:digit:]\.]{1,3} 200" headers.txt && break
+            grep -q -E "HTTP/[[:digit:]\.]{1,3} 429" && echo "$ERR_429"
+            grep -q -E "HTTP/[[:digit:]\.]{1,3} 409" && echo "$ERR_409"
+
+            sleep 30
+          done
+          
+          ENV=$(cat metadata.json | jq -r '.name')
+          echo "::set-output name=environment-name::${ENV}"
+
+      - name: 'Upload Metadata'
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ steps.claim-toolsmiths-env.outputs.environment-name }}
+          path: metadata.json
+
+      - name: Checkout cli-ci
+        uses: actions/checkout@v3
+        with:
+          repository: cloudfoundry/cli-ci
+          path: cli-ci
+
+      - name: Install Tools
+        run: |
+          wget https://github.com/cloudfoundry/bosh-bootloader/releases/download/v8.4.110/bbl-v8.4.110_linux_x86-64 -P /tmp
+          mv /tmp/bbl-* /usr/local/bin/bbl
+          chmod +x /usr/local/bin/bbl
+
+          wget https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip -P /tmp
+          unzip -d /tmp /tmp/awscli-exe-linux-x86_64.zip
+          ./tmp/aws/install
+
+          wget https://s3.amazonaws.com/bosh-cli-artifacts/bosh-cli-7.0.1-linux-amd64 --output-document="/usr/bin/bosh"
+          chmod +x /usr/bin/bosh
+
+      - name: Deploy edge CAPI
+        if: ${{ inputs.capi-version == 'edge' }}
+        run: |
+          # find latest capi
+          FILENAME="$(aws s3 ls capi-releases --no-sign-request --recursive --region us-east-1 | sort | tail -n 1 | awk '{print $4}')"
```

Comment:
I assume this means it is possible to have the same release version and we're just grabbing the latest? Or that since this is edge just always grab the latest and we don't actually know the latest version of capi

Reply:
We don't know what the latest edge is, we just grab it. This is from the output of [capi-create-release](https://ci.cake.capi.land/teams/main/pipelines/capi/jobs/create-capi-release), which passed unit tests but has not passed integration tests

Future PRs will allow specification of CAPI, to support running tests with a minimum supported version of CAPI